fixtures/load-repo.js

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env node
+const kleur = require('kleur');
+const path = require('path');
+const { spawn } = require('child_process');
+const mkdir = require('mkdirplz');
+const rm = require('eliminate');
+const walk = require('powerwalker');
+const { promisify } = require('util');
+const { readFile, writeFile, existsSync } = require('fs');
+const readFileAsync = promisify(readFile);
+const writeFileAsync = promisify(writeFile);
+const babel = require('@babel/core');
+const CLIEngine = require('eslint').CLIEngine;
+const prettier = require('prettier');
+const prompts = require('prompts');
+
+const fixturesDir = __dirname;
+const reposDir = path.join(fixturesDir, 'repos');
+const prettierOpts = {
+  parser: 'babel',
+  ...require('../package.json').prettier,
+};
+
+/**
+ * Wraps child_process.spawn to make it promise-friendly and output to stderr/stdout
+ * @param {string} command
+ * @param {readonly string[]} args
+ * @param {import("child_process").SpawnOptionsWithoutStdio} [opts]
+ */
+const runCommand = (command, args, opts) =>
+  new Promise((resolve, reject) => {
+    const child = spawn(command, args, {
+      ...opts,
+      stdio: [process.stdin, process.stdout, process.stderr],
+    });
+    child.on('exit', (code) => {
+      if (code === 0) resolve();
+      else reject(code);
+    });
+  });
+
+/**
+ * @param {string} name
+ */
+const getRepoDir = (name) => path.join(reposDir, name);
+
+/**
+ * @param {string} name
+ * @param {string} url
+ */
+const cloneRepo = async (name, url) => {
+  console.log(kleur.bold().blue(`Cloning repo: ${name} from ${url}`));
+  // Clear out all existing repos
+  if (existsSync(reposDir)) await rm(reposDir);
+  await mkdir(reposDir);
+  await runCommand('git', ['clone', url, getRepoDir(name)]);
+  console.log(kleur.bold().blue(`Done cloning repo: ${name}`));
+};
+
+const main = async () => {
+  const { url } = await prompts({
+    name: 'url',
+    type: 'text',
+    message: 'GitHub Repo URL',
+  });
+  const name = url.replace(/\.git$/, '').replace(/.*\//, '');
+  await cloneRepo(name, url);
+  const files = await walk(name, { filesonly: true, cwd: reposDir });
+
+  console.log(kleur.bold().blue('Looking for files to lint'));
+
+  const interestingFiles = await Promise.all(
+    files
+      .filter((f) => f.endsWith('.js') && !f.replace(name, '').includes('test'))
+      .map(async (file) => {
+        const contents = await readFileAsync(path.join(reposDir, file), 'utf8');
+        return { file, contents };
+      })
+  );
+
+  const eslint = new CLIEngine({
+    configFile: path.join(fixturesDir, '.eslintrc.js'),
+    fix: true,
+  });
+
+  const filesWithNumNodeTypes = await Promise.all(
```

Comment:
Nit: This would be a little easier to follow if broken up more, as possible. E.g, move the anonymous async function within `map` to be on its own, and then have only one line inside the `Promise.all`.

Reply:
Yeah, I think you are right. I struggled with the organization of this a lot and kept rearranging it. I'll try out what you suggested

Reply:
I realize there are tradeoffs to breaking things out, so just look at where it makes sense and doesn't necessitate excessive rewriting.